tests/atomic-test.lisp

```diff
@@ -17,10 +17,14 @@
 
 (test swap-reference
   "Swap a reference."
-  (let* ((ref (make-atomic-reference :value '()))
-         (old (atomic-get ref)))
-    (is-true (atomic-cas ref old '(1 2)))
-    (is (equal '(1 2) (atomic-get ref)))))
+  (let* ((ref (make-atomic-reference :value '(0)))
+         (fn1 #'append)
+         (args (list 1 2))
+         (fn2 (lambda (lst) (apply #'+ lst))))
+    (is (equal '(0 1 2) (cl-gserver.atomic:atomic-swap ref fn1 args))) ; test with rest args
```

Comment:
Wouldn't just `atomic-swap` work in parallel to just atomic-get`?

Reply:
`atomic-get` is just an accessor of the atomic object and it can't guarantee the value return is not modified by other thread, even though `sbcl` can insert a barrier there, I think it's not a good idea to read and modify to the same place at the same time. Luckly, `cl-gserver` invokes `atomic-get` only in the method of actors in actor-context.lisp, and it will not conflict with `atomic-swap` in other functions.

Reply:
Sorry, I wasn't clear.
What I mean is: can't we just write `atomic-swap` instead of the longer form `cl-gserver.atomic:atomic-swap`, because the package `cl-gserver.atomic` is already known here, or?

Reply:
Sorry, I still cannot make sure how we just write `atomic-swap`. 
If we just write `atomic-swap` without its package name, we have two ways: 
1. import the `atomic-swap` symbol in the packages defining where they want to use this function.
2. `(:use :cl-gserver.atomic)` in the packages defining to intern all symbols of the package `cl-gserver.atomic`.
Do you mean one of those?

By the way, there were many  defpackages in `cl-gserver` and each package has its short nickname, if `cl-gserver` is used in a large system, the nicknames may easy to get symbol conflict with the names of that large system's own packages, especially the commonly used package names such as `utils`. CL does not support such package importing alias feature like `import numpy as np`, when this happens, it' s really a trouble.

Reply:
On the top of the file we have `(:use :cl :fiveam :cl-gserver.atomic)`. So the whole package `:cl-gserver.atomic` is imported (inherited). With that, and considering `atomic-swap` is exported in that package it should be possible to just write `atomic-swap`. Or I completely misunderstand something here.

Regarding the package names, you may be right. `util` may be a problem, the others not so much. Most CL implementations support package-local nicknames. But even if not, a solution can be found I think if the problem arises.